substrate/frame/system/src/offchain.rs

```diff
@@ -75,34 +76,18 @@ pub struct ForAny {}
 /// For submitting unsigned transactions, `submit_unsigned_transaction`
 /// utility function can be used. However, this struct is used by `Signer`
 /// to submit a signed transactions providing the signature along with the call.
-pub struct SubmitTransaction<T: SendTransactionTypes<OverarchingCall>, OverarchingCall> {
+pub struct SubmitTransaction<T: CreateTransactionBase<OverarchingCall>, OverarchingCall> {
```

Comment:
tbh I would also treat this file with a grain of salt and consider the fact that there might be superfluous code in it. I always remember looking at it and thinking it is over-engineered :) maybe you can simplify it a bit.

Reply:
I had the same gut feeling and it's mostly because of the (soon to be) legacy interface of signed transactions, namely keeping `create_signed_transaction` backwards compatible. It's part of an offchain-facing interface which uses the public key, address and nonce along with the call to create a transaction, which is obviously not generic at all and needs these types somewhere. Once all of these are abstracted away in a new-school transaction extension, the interface should get a lot leaner and cleaner.

Also, for this transition period, I opted to use a "base" trait to unify the call and extrinsic types. This should prevent users from mistakenly using different types in the child traits and also reduce boilerplate code a bit.